### PR TITLE
Improve add_dag_code_table migration

### PR DESCRIPTION
**Pull request by ashb**
This migration was added in #7217 had a few problems that could cause us trouble in the
future:

1. It was importing the live model definition, which could break in the
   future (if for instance it ever got a new column added, when this
   migration runs it would try to use that new definition, but the
   column won't exist yet).
2. It was selecting the (large) `data` column needlessly
3. It was dropping and re-creating the index but that is only needed on
   MSSQL, not for MySQL or Postgresql.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: airflow/migrations/versions/952da73b5eff_add_dag_code_table.py [hunk 1]
```diff
@@ -38,6 +37,18 @@
...
+
+    Base = declarative_base()
+
+    class SerializedDagModel(Base):
+        __tablename__ = 'serialized_dag'
+
+        # There are other columns here, but these are the only ones we need for the SELECT/UPDATE we are doing
+        dag_id = sa.Column(sa.String(250), primary_key=True)
+        fileloc = sa.Column(sa.String(2000), nullable=False)
+        fileloc_hash = sa.Column(sa.BigInteger, nullable=False)
```
Comment (kaxil) [inline airflow/migrations/versions/952da73b5eff_add_dag_code_table.py:50]: Should the column type be old one here?

Comment (kaxil) [reply airflow/migrations/versions/952da73b5eff_add_dag_code_table.py:50]: i.e. `Integer` ?

Comment (kaxil) [reply airflow/migrations/versions/952da73b5eff_add_dag_code_table.py:50]: Ignore me. I see we use the model after changing Column type

Comment (ashb) [reply airflow/migrations/versions/952da73b5eff_add_dag_code_table.py:50]: Yup.

**Pull request merged**